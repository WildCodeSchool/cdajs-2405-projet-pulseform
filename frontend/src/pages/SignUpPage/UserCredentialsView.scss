@use "@assets/variables" as *;

.signup-page {
  &__logo {
    display: none;

    @media screen and (min-width: 1023px) {
      display: flex;
    }
  }

  &__blob {
    position: absolute;
    width: 85%;
    top: 0;
    right: 0;
    z-index: 0;
    overflow: hidden;

    @media screen and (min-width: 1023px) {
      display: none;
    }
    @media screen and (min-width: 795px) {
      width: 20%;
    }
    @media screen and (min-width: 460px) {
      width: 40%;
    }
  }

  &__form-section {
    display: flex;
    flex-direction: column;
    align-items: center;
    min-height: 100vh;

    @media screen and (min-width: 1023px) {
      flex-direction: row-reverse;
      justify-content: center;
      align-items: center;
      background-color: $white;
    }
  }

  &__form-container {
    background-color: $primary;
    padding: 4% 8%;
    margin-left: 5rem;
    border-radius: 10px;
    box-shadow: 50px 50px 50px rgba(0, 0, 0, 0.3);

    @media screen and (max-width: 1023px) {
      background-color: transparent;
      box-shadow: none;
      padding: 2rem 1.5rem;
      margin: 6rem auto 0;
    }
    @media screen and (max-width: 1223px) {
      padding: 3% 6%;
      margin-left: 20%;
    }
  }

  &__ready-title {
    display: block;
    @include h1-phone;
    color: $primary;
    text-align: center;
    margin-top: 4rem;

    @media screen and (min-width: 1023px) {
      display: none;
    }
  }

  &__title {
    @include h1-phone;
    color: $white;
    margin-bottom: 2rem;
    text-align: center;
    font-size: 1.3rem;

    @media screen and (min-width: 1023px) {
      text-align: center;
    }
  }

  &__form {
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 0.8rem;

    @media screen and (min-width: 1023px) {
      align-items: stretch;
    }
  }

  &__error-message {
    color: red;
    font-size: 0.7rem;
    text-align: center;
    max-height: 0.05rem;
    transition: opacity 0.2s ease;

    @media screen and (min-width: 1023px) {
      text-align: left;
      max-height: 0.05rem;
    }
  }

  &__connect-button {
    background-color: $secondary;
    color: $white;
    border: none;
    border-radius: 5px;
    cursor: pointer;
    font-size: 1rem;
    margin: 4rem 0 2rem;
    margin-top: 5rem;
    width: 100%;
    max-width: 20rem;
    align-self: center;

    @media screen and (min-width: 1023px) {
      margin-top: 4rem;
    }
  }

  &__image-side {
    display: none;

    @media screen and (min-width: 1023px) {
      display: flex;
    }
  }

  &__step-image {
    width: 85%;
    height: auto;
    box-shadow: 50px 50px 50px rgba(0, 0, 0, 0.3);
  }

  &__test-program-container {
    display: none;

    @media screen and (min-width: 1023px) {
      display: flex;
      justify-content: center;
      align-items: center;
      position: absolute;
      width: 100%;
      top: 7%;
      z-index: 2;
    }
  }

  &__test-program-button {
    @include h1-phone;
    padding: 10px 20px;
    font-size: 18px;
    color: $primary;
    background-color: transparent;
    border: 2px solid black;
    border-radius: 10px;
    cursor: pointer;
  }
}

.mobile-only {
  display: block;

  @media screen and (min-width: 1023px) {
    display: none;
  }
}
