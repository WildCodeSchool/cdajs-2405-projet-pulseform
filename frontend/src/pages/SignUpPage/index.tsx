import { useState } from "react";
import { useForm } from "react-hook-form";
import { useTranslation } from "react-i18next";
import { Link } from "react-router-dom";

import blopLoginPage from "@assets/icons/blob/blob3.svg";
import BasicButton from "@components/atoms/BasicButton";
import InputField from "@components/atoms/ImputField/ImputField";
import BodyStepQuestions from "@components/molecules/BodyStepQuestions";
import UserInfoAddView from "@components/molecules/UserInfoAddView";

import "./UserCredentialsView.scss";

interface FormData {
  email: string;
  password: string;
  confirmPassword: string;
}

function UserCredentialsView() {
  const { register, handleSubmit, setError, formState } = useForm<FormData>();
  const { t } = useTranslation();
  const [isNextStep, setIsNextStep] = useState(false);

  const onSubmit = (data: FormData) => {
    if (data.password !== data.confirmPassword) {
      setError("confirmPassword", {
        type: "manual",
        message: t("PASSWORDS_NOT_MATCHING"),
      });
      return;
    }

    setIsNextStep(true);
  };

  if (isNextStep) return <UserInfoAddView />;

  return (
    <div className="signup-page">
      <section className="signup-page__form-section signup-page__mobile">
        <img
          className="signup-page__blob"
          src={blopLoginPage}
          alt={t("BLOB_ALT_TEXT")}
        />

        <div className="signup-page__form-container">
          <h1 className="signup-page__title">{t("CREATE_ACCOUNT")}</h1>

          <form className="signup-page__form" onSubmit={handleSubmit(onSubmit)}>
            <InputField<FormData>
              name="email"
              type="email"
              placeholderKey="EMAIL"
              register={register}
              required
              ariaLabel="EMAIL"
            />
            <InputField<FormData>
              name="password"
              type="password"
              placeholderKey="PASSWORD"
              register={register}
              required
              ariaLabel="PASSWORD"
            />
            <InputField<FormData>
              name="confirmPassword"
              type="password"
              placeholderKey="CONFIRM_PASSWORD"
              register={register}
              required
              ariaLabel="CONFIRM_PASSWORD"
            />
            {formState.errors.confirmPassword && (
              <p className="signup-page__error-message">
                {formState.errors.confirmPassword.message as string}
              </p>
            )}

            <BasicButton
              typeButton="orange"
              type="submit"
              className="signup-page__connect-button"
            >
              {t("CREATE_ACCOUNT")}
            </BasicButton>
          </form>
        </div>

        <section className="signup-page__section">
          <div className="signup-page__align">
            <div className="signup-page__motivation-block">
              <p className="signup-page__motivation-title">
                {t("ALREADY_HAVE_ACCOUNT")}
              </p>
              <div className="signup-page__primary-trait" />
            </div>
            <div className="signup-page__create-account-block">
              <Link to="/login" className="signup-page__create-account-button">
                {t("CONNECT")}
              </Link>
              <div className="signup-page__secondary-trait" />
            </div>
          </div>
        </section>
      </section>

      <section className="signup-page__desktop">
        <BodyStepQuestions
          questionLabel={t("CREATE_ACCOUNT")}
          ctaExit={() => {}}
        >
          <form
            className="user-credentials-view__container__form"
            onSubmit={handleSubmit(onSubmit)}
          >
            <InputField<FormData>
              name="email"
              type="email"
              placeholderKey="EMAIL"
              register={register}
              required
              ariaLabel="EMAIL"
            />
            <InputField<FormData>
              name="password"
              type="password"
              placeholderKey="PASSWORD"
              register={register}
              required
              ariaLabel="PASSWORD"
            />
            <InputField<FormData>
              name="confirmPassword"
              type="password"
              placeholderKey="CONFIRM_PASSWORD"
              register={register}
              required
              ariaLabel="CONFIRM_PASSWORD"
            />
            {formState.errors.confirmPassword && (
              <p className="signup-page__error-message">
                {formState.errors.confirmPassword.message as string}
              </p>
            )}
            <BasicButton
              className="user-credentials-view__container__form__btn"
              type="submit"
              typeButton="orange"
            >
              {t("CREATE_ACCOUNT")}
            </BasicButton>
          </form>
        </BodyStepQuestions>
      </section>
    </div>
  );
}

export default UserCredentialsView;
