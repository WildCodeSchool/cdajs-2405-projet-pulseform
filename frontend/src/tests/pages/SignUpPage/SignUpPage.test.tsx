import { MockedProvider } from "@apollo/client/testing";
import SignUpPage from "@pages/SignUpPage";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

describe("SignUpPage", () => {
  it("renders the SignUp Page", () => {
    const tree = render(
      <MemoryRouter>
        <MockedProvider mocks={[]} addTypename={false}>
          <SignUpPage />
        </MockedProvider>
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
