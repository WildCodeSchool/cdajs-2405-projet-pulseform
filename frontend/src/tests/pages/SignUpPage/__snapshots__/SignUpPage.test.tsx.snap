// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignUpPage > renders the SignUp Page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="little-logo signup-page__logo little-logo--desktop"
      >
        <img
          alt="Logo Pulse Form"
          class="little-logo__img"
          src="/src/assets/icons/mobile-logo.svg"
        />
        <p
          class="little-logo__title little-logo__title--desktop"
        >
          Pulse Form
        </p>
      </div>
      <img
        alt="BLOB_ALT_TEXT"
        aria-hidden="true"
        class="signup-page__blob"
        src="/src/assets/icons/blob/blob3.svg"
      />
      <section
        class="signup-page__form-section"
      >
        <div
          class="signup-page__image-side desktop-only"
        >
          <img
            alt="STEP_IMAGE_ALT"
            class="signup-page__step-image"
            src="/src/assets/images/step-img.svg"
          />
        </div>
        <div
          class="signup-page__form-container"
        >
          <h1
            class="login-page__title signup-page__ready-title mobile-only"
          >
            CREATE_ACCOUNT
          </h1>
          <h1
            class="signup-page__title"
          >
            CREATE_ACCOUNT
          </h1>
          <form
            class="signup-page__form"
          >
            <div
              class="input-field-wrapper"
            >
              <input
                aria-label="EMAIL"
                aria-required="true"
                class="input__login-page"
                name="email"
                placeholder="EMAIL"
                type="email"
                value=""
              />
            </div>
            <div
              class="input-field-wrapper"
            >
              <input
                aria-label="USERNAME"
                aria-required="true"
                class="input__login-page"
                name="username"
                placeholder="USERNAME"
                type="text"
                value=""
              />
            </div>
            <div
              class="input-field-wrapper"
            >
              <input
                aria-label="PASSWORD"
                aria-required="true"
                class="input__login-page"
                name="password"
                placeholder="PASSWORD"
                type="password"
                value=""
              />
              <button
                aria-label="SHOW_PASSWORD"
                class="input-field-wrapper__toggle "
                type="button"
              >
                <span />
              </button>
            </div>
            <div
              class="input-field-wrapper"
            >
              <input
                aria-label="CONFIRM_PASSWORD"
                aria-required="true"
                class="input__login-page"
                name="confirmPassword"
                placeholder="CONFIRM_PASSWORD"
                type="password"
                value=""
              />
              <button
                aria-label="SHOW_PASSWORD"
                class="input-field-wrapper__toggle "
                type="button"
              >
                <span />
              </button>
            </div>
            <button
              aria-label="CREATE_ACCOUNT"
              class="basic-button signup-page__connect-button basic-button--orange"
              type="submit"
            >
              CREATE_ACCOUNT
            </button>
          </form>
          <section
            aria-label="ACCOUNT_ACTIONS"
            class="already-member"
          >
            <div
              class="already-member__align"
            >
              <div
                class="already-member__motivation-block"
              >
                <p
                  class="already-member__motivation-title"
                >
                  ALREADY_HAVE_ACCOUNT
                </p>
                <div
                  class="already-member__primary-trait"
                />
              </div>
              <div
                class="already-member__create-account-block"
              >
                <a
                  aria-label="GO_TO_LOGIN"
                  class="already-member__create-account-button"
                  href="/login"
                >
                  CONNECT
                </a>
                <div
                  class="already-member__secondary-trait"
                />
              </div>
            </div>
          </section>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <div
      class="little-logo signup-page__logo little-logo--desktop"
    >
      <img
        alt="Logo Pulse Form"
        class="little-logo__img"
        src="/src/assets/icons/mobile-logo.svg"
      />
      <p
        class="little-logo__title little-logo__title--desktop"
      >
        Pulse Form
      </p>
    </div>
    <img
      alt="BLOB_ALT_TEXT"
      aria-hidden="true"
      class="signup-page__blob"
      src="/src/assets/icons/blob/blob3.svg"
    />
    <section
      class="signup-page__form-section"
    >
      <div
        class="signup-page__image-side desktop-only"
      >
        <img
          alt="STEP_IMAGE_ALT"
          class="signup-page__step-image"
          src="/src/assets/images/step-img.svg"
        />
      </div>
      <div
        class="signup-page__form-container"
      >
        <h1
          class="login-page__title signup-page__ready-title mobile-only"
        >
          CREATE_ACCOUNT
        </h1>
        <h1
          class="signup-page__title"
        >
          CREATE_ACCOUNT
        </h1>
        <form
          class="signup-page__form"
        >
          <div
            class="input-field-wrapper"
          >
            <input
              aria-label="EMAIL"
              aria-required="true"
              class="input__login-page"
              name="email"
              placeholder="EMAIL"
              type="email"
              value=""
            />
          </div>
          <div
            class="input-field-wrapper"
          >
            <input
              aria-label="USERNAME"
              aria-required="true"
              class="input__login-page"
              name="username"
              placeholder="USERNAME"
              type="text"
              value=""
            />
          </div>
          <div
            class="input-field-wrapper"
          >
            <input
              aria-label="PASSWORD"
              aria-required="true"
              class="input__login-page"
              name="password"
              placeholder="PASSWORD"
              type="password"
              value=""
            />
            <button
              aria-label="SHOW_PASSWORD"
              class="input-field-wrapper__toggle "
              type="button"
            >
              <span />
            </button>
          </div>
          <div
            class="input-field-wrapper"
          >
            <input
              aria-label="CONFIRM_PASSWORD"
              aria-required="true"
              class="input__login-page"
              name="confirmPassword"
              placeholder="CONFIRM_PASSWORD"
              type="password"
              value=""
            />
            <button
              aria-label="SHOW_PASSWORD"
              class="input-field-wrapper__toggle "
              type="button"
            >
              <span />
            </button>
          </div>
          <button
            aria-label="CREATE_ACCOUNT"
            class="basic-button signup-page__connect-button basic-button--orange"
            type="submit"
          >
            CREATE_ACCOUNT
          </button>
        </form>
        <section
          aria-label="ACCOUNT_ACTIONS"
          class="already-member"
        >
          <div
            class="already-member__align"
          >
            <div
              class="already-member__motivation-block"
            >
              <p
                class="already-member__motivation-title"
              >
                ALREADY_HAVE_ACCOUNT
              </p>
              <div
                class="already-member__primary-trait"
              />
            </div>
            <div
              class="already-member__create-account-block"
            >
              <a
                aria-label="GO_TO_LOGIN"
                class="already-member__create-account-button"
                href="/login"
              >
                CONNECT
              </a>
              <div
                class="already-member__secondary-trait"
              />
            </div>
          </div>
        </section>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
